### PR TITLE
Nc feature auth

### PR DESCRIPTION
**Pull request by nancychuchu**
Set up navigation with initial route at the Auth stack which will check if there's a user token. Sign in and register is being addressed in a different issue. In the meantime, clicking sign in will redirect to the app and logout will bring back to the Auth Welcome screen.

### CODE REVIEW

File: client/js/navigation/NavigationLayout.js [hunk 1]
```diff
@@ -139,6 +139,20 @@ export default createDrawerNavigator(
   },
   {
     contentComponent: props => <DrawerContent {...props} />,
+    contentComponent: props => (
+      <View style={{ flex: 1 }}>
+        <SafeAreaView forceInset={{ top: "always", horizontal: "never" }}>
```
Comment (siddharthparmar7) [inline client/js/navigation/NavigationLayout.js:147]: Do you think your `SafeAreaView` should be the parent with `flex:1` and View component can be the child?

File: client/js/screens/AuthWelcome/AuthWelcome.js [hunk 2]
```diff
@@ -1,6 +1,14 @@
...
+import {
+  View,
+  Image,
+  Button,
+  ImageBackground,
+  Text,
+  TouchableOpacity
+} from "react-native";
 import styles from "./styles";
+// import { withNavigation } from "react-navigation";
```
Comment (siddharthparmar7) [inline client/js/screens/AuthWelcome/AuthWelcome.js:11]: Please remove commented code.

File: client/js/screens/AuthWelcome/AuthWelcomeContainer.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React, { Component } from "react";
+import AuthWelcome from "./AuthWelcome";
+class AuthWelcomeContainer extends Component {
+  static navigationOptions = {
+    title: "SignIn"
```
Comment (siddharthparmar7) [inline client/js/screens/AuthWelcome/AuthWelcomeContainer.js]: There is no header so you do not need a title for this screen. Please remove it.

File: client/js/screens/AuthWelcome/styles.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import { StyleSheet, Dimensions } from "react-native";
+import globalStyles from "../../config/globalStyles";
+
+const width = Dimensions.get("window").width;
+const height = Dimensions.get("window").height;
```
Comment (siddharthparmar7) [inline client/js/screens/AuthWelcome/styles.js]: suggestion: `const {height, width} = Dimensions.get("window");`

File: client/js/screens/Login/Login.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  handleInput = () => {
+    console.log("something submitted");
+  };
+
+  goToRegister = () => {
+    this.props.navigation.navigate("Register");
+  };
+
+  _signInAsync = async () => {
+    await AsyncStorage.setItem("userToken", "abc");
```
Comment (siddharthparmar7) [inline client/js/screens/Login/Login.js]: create a model.js file where you can store all your DB related methods and call them from anywhere in your app by importing them.

File: client/js/screens/Login/Login.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+                      onSubmit={handleSubmit}
+                      editable={true}
+                      secureTextEntry={true}
+                    />
+                  </View>
+                )}
+              />
+
+              <Text>Forgot your password?</Text>
+
```
Comment (siddharthparmar7) [inline client/js/screens/Login/Login.js:44]: Please show user validation errors `(ex:) User not found. User or Password incorrect.`

File: client/js/screens/Login/styles.js [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  labelText: {
+    padding: 10,
+    color: globalStyles.mediumGreen,
+    alignSelf: "flex-start"
+  },
+  formInput: {
+    borderStyle: "solid",
+    borderWidth: 2,
+    borderColor: globalStyles.lightGrey,
+    // width: width * 0.8,
```
Comment (siddharthparmar7) [inline client/js/screens/Login/styles.js]: Please remove commented code.

File: client/js/screens/Register/Register.js [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+                render={({ input, meta }) => (
+                  <View style={styles.formInput}>
+                    <TextInput
+                      onSubmit={handleSubmit}
+                      editable={true}
+                      secureTextEntry={true}
+                    />
+                  </View>
+                )}
+              />
```
Comment (siddharthparmar7) [inline client/js/screens/Register/Register.js]: Show auth errors.

File: client/js/screens/Register/RegisterContainer.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import React, { Component } from "react";
+import { Text } from "react-native";
+import Register from "./Register";
+
+class RegisterContainer extends Component {
+  static navigationOptions = {
+    title: "Register"
+  };
```
Comment (siddharthparmar7) [inline client/js/screens/Register/RegisterContainer.js]: There is no header for this screen. Remove title.

File: client/js/screens/Register/Register.js [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+  Text,
+  Image
+} from "react-native";
+import { Form, Field } from "react-final-form";
+import styles from "./styles";
+
+class Register extends Component {
+  static navigationOptions = {
+    title: "Register"
+  };
```
Comment (siddharthparmar7) [inline client/js/screens/Register/Register.js]: There is no header for this screen. Remove title.

File: client/js/screens/Login/LoginContainer.js [hunk 11]
```diff
@@ -0,0 +1,15 @@
+import React, { Component } from "react";
+import { Text } from "react-native";
+import Login from "./Login";
+
+class LoginContainer extends Component {
+  static navigationOptions = {
+    title: "Login"
+  };
```
Comment (siddharthparmar7) [inline client/js/screens/Login/LoginContainer.js]: There is no header for this screen. Remove title.

File: client/js/screens/Login/Login.js [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+  Image,
+  TextInput
+} from "react-native";
+import { Form, Field } from "react-final-form";
+import styles from "./styles";
+
+class Login extends Component {
+  static navigationOptions = {
+    title: "Login"
+  };
```
Comment (siddharthparmar7) [inline client/js/screens/Login/Login.js]: There is no header for this screen. Remove title.

File: client/js/screens/Register/styles.js [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+  labelText: {
+    padding: 10,
+    color: globalStyles.mediumGreen,
+    alignSelf: "flex-start"
+  },
+  formInput: {
+    borderStyle: "solid",
+    borderWidth: 2,
+    borderColor: globalStyles.lightGrey,
+    // width: width * 0.8,
```
Comment (siddharthparmar7) [inline client/js/screens/Register/styles.js]: Please remove commented code.

File: yarn.lock [hunk 14]
```diff
@@ -0,0 +1,4 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (siddharthparmar7) [inline yarn.lock]: Remove this file.

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Please make the changes. Thank you.

File: client/js/navigation/NavigationLayout.js [hunk 15]
```diff
@@ -139,6 +139,20 @@ export default createDrawerNavigator(
...
+          <DrawerItems {...props} />
+          <Button
+            title="Logout"
+            onPress={async () => {
+              await AsyncStorage.clear();
+              props.navigation.navigate("AuthLoading");
```
Comment (teelau) [inline client/js/navigation/NavigationLayout.js]: Add error handling for when AsyncStorage or navigation produces an error. Try catch block can fix this.

**Review by teelau — CHANGES_REQUESTED** [review]

File: client/js/navigation/NavigationLayout.js (hunk 1 above)
Comment (nancychuchu) [reply client/js/navigation/NavigationLayout.js:147]: I tried this but it throw the LOGOUT button to the top of the menu and all the other links disappeared.

File: client/js/screens/Register/Register.js (hunk 8 above)
Comment (nancychuchu) [reply client/js/screens/Register/Register.js]: Taking care of this in a separate issue.

File: client/js/screens/Login/Login.js (hunk 6 above)
Comment (nancychuchu) [reply client/js/screens/Login/Login.js:44]: Taking care of this in a separate issue.

**Comment by nancychuchu** [comment]
Validation and signIn/Register functionality will be implemented in another issue.

File: client/js/screens/AuthLoading/AuthLoadingContainer.js [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+} from "react-native";
+import AuthLoading from "./AuthLoading";
+class AuthLoadingContainer extends Component {
+  constructor(props) {
+    super(props);
+    this._bootstrapAsync();
+  }
+
+  _bootstrapAsync = async () => {
+    const userToken = await AsyncStorage.getItem("userToken");
```
Comment (siddharthparmar7) [inline client/js/screens/AuthLoading/AuthLoadingContainer.js]: Add try catch block around all your db calls.

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]

**Review by siddharthparmar7 — APPROVED** [review]

**Pull request merged**